test/jasmine/tests/range_slider_test.js

```diff
@@ -0,0 +1,229 @@
+
+            var maskMax = children[3],
+                handleMax = children[6];
+
+            expect(rangeSlider.getAttribute('data-max')).toEqual(String(+dataMaxStart + diff));
+            expect(maskMax.getAttribute('width')).toEqual(String(-diff));
+            expect(handleMax.getAttribute('transform')).toBe('translate(' + (+dataMaxStart + diff) + ')');
+        });
+
+        it('should react to moving the slidebox left to right', function() {
+            var start = 250,
+                end = 300,
+                dataMinStart = rangeSlider.getAttribute('data-min'),
+                diff = end - start;
+
+            slide(start, 400, end, 400);
+
+            var maskMin = children[2],
+                handleMin = children[5];
+
+            expect(rangeSlider.getAttribute('data-min')).toEqual(String(+dataMinStart + diff));
+            expect(maskMin.getAttribute('width')).toEqual(String(diff));
+            expect(handleMin.getAttribute('transform')).toEqual('translate(' + (+dataMinStart + diff - 3) + ')');
+        });
+
+        it('should react to moving the slidebox right to left', function() {
+            var start = 300,
+                end = 250,
+                dataMaxStart = rangeSlider.getAttribute('data-max'),
+                diff = end - start;
+
+            slide(start, 400, end, 400);
+
+            var maskMax = children[3],
+                handleMax = children[6];
+
+            expect(rangeSlider.getAttribute('data-max')).toEqual(String(+dataMaxStart + diff));
+            expect(maskMax.getAttribute('width')).toEqual(String(-diff));
+            expect(handleMax.getAttribute('transform')).toEqual('translate(' + (+dataMaxStart + diff) + ')');
+        });
+
+        it('should resize the main plot when rangeslider has moved', function() {
+            var start = 300,
+                end = 400,
+                rangeDiff1 = gd._fullLayout.xaxis.range[1] - gd._fullLayout.xaxis.range[0];
+
+            slide(start, 400, end, 400);
+
+            var rangeDiff2 = gd._fullLayout.xaxis.range[1] - gd._fullLayout.xaxis.range[0];
+
+            expect(rangeDiff2).toBeLessThan(rangeDiff1);
+
+            start = 400;
+            end = 200;
+
+            slide(start, 400, end, 400);
+
+            var rangeDiff3 = gd._fullLayout.xaxis.range[1] - gd._fullLayout.xaxis.range[0];
+
+            expect(rangeDiff3).toBeLessThan(rangeDiff2);
+        });
+    });
+
+
+    describe('when not specified as visible', function() {
+        beforeEach(function(done) {
+            gd = createGraphDiv();
+
+            Plotly.plot(gd, [{ x: [1,2,3], y: [2,3,4] }], {}).then(function() {
+                rangeSlider = document.getElementsByClassName('range-slider')[0];
+                done();
+            });
+        });
+
+        afterEach(destroyGraphDiv);
+
+        it('should not be added to the DOM by default', function() {
+            expect(rangeSlider).not.toBeDefined();
+        });
+    });
```

Comment:
Great test.

Let's add one more that checks that `Plotly.relayout(gd, 'xaxis.rangeslider', false)` does clear the slider.